### DISCUSSION TITLE
Something went wrong :/ list index out of range

### DISCUSSION THREAD

**Discussion by talhaanwarch** [open-webui/open-webui#4874] [Q&A]
Hi, I am trying to use my tool locally and getting the error `Something went wrong :/ list index out of range

How to resolve the issue.

**Comment by talhaanwarch** [comment]
I think i missed doc string and type hint in required format

**Reply by lotherk** [reply]
why don't you share what was wrong with yours...?

**Comment by Jan-Jaap** [comment]
I've had this same error. Tools server connection was fine, but a toolcall in a chat resulted in "List index out of range"
Turned out I had two tool server connections pointing to the same API defined in Admin > Settings > Tools.
Removing one fixed the issue.

I was messing with docker networking and had one accessing the tool server through host network and one accessing the docker container directly.

**Reply by AldenWangExis** [reply]
Hi, I'm experiencing the same issue, but it's still **unresolved**. Could you assist?

Here are my findings:

1. Verified the model is working properly.
2. Verified the Tools server works independently (mcp_config tested via windsurf and /docs endpoint).
3. When adding tools to the model, the frontend throws "list index out of range" error.

Troubleshooting attempted:
- Followed your suggestion to remove generic APIs and keep only one Tools server
- Issue persists after these changes

Environment:
- OpenWebUI version: 0.6.2
- MCPO launched using Python (mcpo --config /path/to/config.json --port 8182)
- Tools server http://xxxxxx.com:8182/postgres w/o api-key

Backend logs:
2025-04-14 15:55:00

> .216 | INFO     | uvicorn.protocols.http.httptools_impl:send:476 - xxx.xxx.xx.134:59694 - "GET /api/v1/users/user/settings HTTP/1.1" 200 - {}
> 2025-04-14 15:55:05.886 | INFO     | uvicorn.protocols.http.httptools_impl:send:476 - xxx.xxx.xx.134:59708 - "POST /api/v1/chats/new HTTP/1.1" 200 - {}
> 2025-04-14 15:55:06.047 | INFO     | uvicorn.protocols.http.httptools_impl:send:476 - xxx.xxx.xx.134:59708 - "GET /api/v1/chats/?page=1 HTTP/1.1" 200 - {}
> 2025-04-14 15:55:06.105 | INFO     | uvicorn.protocols.http.httptools_impl:send:476 - xxx.xxx.xx.134:59708 - "OPTIONS /api/v1/chats/1be3205c-2a78-4a87-b211-babd5af0ba68 HTTP/1.1" 200 - {}
> 2025-04-14 15:55:06.207 | INFO     | uvicorn.protocols.http.httptools_impl:send:476 - xxx.xxx.xx.134:59708 - "POST /api/v1/chats/1be3205c-2a78-4a87-b211-babd5af0ba68 HTTP/1.1" 200 - {}
> 2025-04-14 15:55:06.329 | INFO     | uvicorn.protocols.http.httptools_impl:send:476 - xxx.xxx.xx.134:59708 - "GET /api/v1/chats/?page=1 HTTP/1.1" 200 - {}
> 2025-04-14 15:55:06.483 | INFO     | uvicorn.protocols.http.httptools_impl:send:476 - xxx.xxx.xx.134:59708 - "POST /api/chat/completions HTTP/1.1" 400 - {}
> 2025-04-14 15:55:06.612 | INFO     | uvicorn.protocols.http.httptools_impl:send:476 - xxx.xxx.xx.134:59708 - "GET /api/v1/chats/?page=1 HTTP/1.1" 200 - {}
